### DISCUSSION TITLE
Is it possible to set disabled_optimizers with C# API?

### DISCUSSION THREAD

**Discussion by asvyazin** [microsoft/onnxruntime#18802] [API Q&A]
Python API has keyword `disabled_optimizers` in `InferenceSession` constructor but I don't see anything similar in C# API. Is there some way to explicitly disable some optimizers in C#?

**Comment by claeyzre** [comment]
You can pass SessionOptions in the InferenceSession constructor
And you can change the parameters of this session options, one of them is the GraphOptimizationLevel, so you can set it to one of these
```public enum GraphOptimizationLevel
    {
        ORT_DISABLE_ALL = 0,
        ORT_ENABLE_BASIC = 1,
        ORT_ENABLE_EXTENDED = 2,
        ORT_ENABLE_ALL = 99
    }
```
and then pass the session options to the inference session and you're good

**Comment by asvyazin** [comment]
It's not the same. With `disabled_optimizers` you can explicitly disable one or more optimizers by name. With `GraphOptimizationLevel` it's unclear which optimizers exactly will be disabled
